python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -446,6 +446,17 @@ def get_mapping_key(self) -> Optional[str]:
     # asset related methods
     #############################################################################################
 
+    @public
+    @property
+    def asset_key(self) -> AssetKey:
+        """The AssetKey for the current asset. In a multi_asset, use asset_key_for_output instead."""
+        if self.has_assets_def and len(self.assets_def.keys_by_output_name.keys()) > 1:
+            raise DagsterInvariantViolationError(
+                "Cannot call `context.asset_key` in a multi_asset. Use"
```

Comment:
"Cannot call `context.asset_key` in a multi_asset that materializes more than one asset"